tendermint-lite/src/main.rs

```diff
@@ -1,135 +0,0 @@
-    ValidatorSet as _,
-};
-use tendermint::rpc;
-use tendermint::{block::Height, Hash};
-
-use tendermint_lite::{
-    requester::RPCRequester, state::State, store::MemStore, threshold::TrustThresholdOneThird,
-};
-
-use core::future::Future;
-use std::time::{Duration, SystemTime};
-use tokio::runtime::Builder;
-
-// TODO: these should be config/args
-static SUBJECTIVE_HEIGHT: u64 = 1;
-static SUBJECTIVE_VALS_HASH_HEX: &str =
-    "A5A7DEA707ADE6156F8A981777CA093F178FC790475F6EC659B6617E704871DD";
-static RPC_ADDR: &str = "localhost:26657";
-
-// TODO: this should somehow be configurable ...
-static THRESHOLD: &TrustThresholdOneThird = &TrustThresholdOneThird {};
-
-pub fn block_on<F: Future>(future: F) -> F::Output {
-    Builder::new()
-        .basic_scheduler()
-        .enable_all()
-        .build()
-        .unwrap()
-        .block_on(future)
-}
-
-fn main() {
-    // TODO: this should be config
-    let trusting_period = Duration::new(6000, 0);
-
-    // setup requester for primary peer
-    let client = block_on(rpc::Client::new(&RPC_ADDR.parse().unwrap())).unwrap();
-    let req = RPCRequester::new(client);
-    let mut store = MemStore::new();
-
-    let vals_hash =
-        Hash::from_hex_upper(hash::Algorithm::Sha256, SUBJECTIVE_VALS_HASH_HEX).unwrap();
-
-    subjective_init(Height::from(SUBJECTIVE_HEIGHT), vals_hash, &mut store, &req).unwrap();
-
-    loop {
-        let latest = (&req).signed_header(0).unwrap();
-        let latest_peer_height = latest.header().height();
-
-        let latest = store.get(Height::from(0)).unwrap();
-        let latest_height = latest.last_header().header().height();
-
-        // only bisect to higher heights
-        if latest_peer_height <= latest_height {
-            std::thread::sleep(Duration::new(1, 0));
-            continue;
-        }
-
-        println!(
-            "attempting bisection from height {:?} to height {:?}",
-            store
-                .get(Height::from(0))
-                .unwrap()
-                .last_header()
-                .header()
-                .height(),
-            latest_peer_height,
-        );
-
-        let now = &SystemTime::now();
-        lite::verify_and_update_bisection(
-            latest_peer_height,
-            THRESHOLD,
-            &trusting_period,
-            now,
-            &req,
-            &mut store,
-        )
-        .unwrap();
-
-        println!("Succeeded bisecting!");
-
-        // notifications ?
-
-        // sleep for a few secs ?
-    }
-}
-
-/*
- * The following is initialization logic that should have a
- * function in the lite crate like:
- * `subjective_init(height, vals_hash, store, requester) -> Result<(), Error`
- * it would fetch the initial header/vals from the requester and populate a
- * trusted state and store it in the store ...
- * TODO: this should take traits ... but how to deal with the State ?
- * TODO: better name ?
- */
```

Comment:
@ebuchman @Shivani912 I'll move this into the lite crate in a separate PR. Note: we need to rework the spec to match the initialization via a pair of subjective (height, vals_hash). Currently, it only talks about init. through a header.